### PR TITLE
#1072 Fixed transpose_optimizer shape assumptions

### PR DESCRIPTION
**Pull request by NikolasMarkou**
By adding additional checks before entering the change phase. There is no change in logic or optimization

### CODE REVIEW

File: tf2onnx/optimizer/transpose_optimizer.py [hunk 1]
```diff
@@ -101,8 +101,8 @@ def _calculate_new_shape(graph, op):
                 input_shape = self._g.get_shape(op.input[0])
                 if not input_shape:
                     continue
-
-                if (is_nchw_transpose(op) and (input_shape[3] == 1 or (input_shape[1:3] == [1, 1]))) \
+                if (is_nchw_transpose(op) and len(input_shape) == 4 and
```
Comment (TomWildenhain-Microsoft) [inline tf2onnx/optimizer/transpose_optimizer.py:104]: Doesn't `is_nchw_transpose` already ensure the length of the input shape is 4?

Comment (NikolasMarkou) [reply tf2onnx/optimizer/transpose_optimizer.py:104]: You are right it does

Comment (TomWildenhain-Microsoft) [reply tf2onnx/optimizer/transpose_optimizer.py]: I take that back.  It seems like the whole issue with #1072 is that somehow a transpose node has perm and shape properties with different dimensions.

Comment (NikolasMarkou) [reply tf2onnx/optimizer/transpose_optimizer.py]: That somehow was the bug i saw, but when reading the code the check seemed redundant, i put it there again.

**Comment by NikolasMarkou** [comment]
@TomWildenhain-Microsoft what is the requested change here ?

**Comment by NikolasMarkou** [comment]
@guschmue

**Comment by TomWildenhain-Microsoft** [comment]
In #1072 I mentioned that this check is only needed if the perms attribute of the transpose is longer than its shape would allow.  We could merge this as a temporary fix but I'm still wondering why this situation is occurring.  It suggests the true bug is somewhere else.

**Review by TomWildenhain-Microsoft — APPROVED** [review]

**Pull request closed without merging**